include/msgpack/v1/adaptor/cpp20/span.hpp

```diff
@@ -0,0 +1,115 @@
+            }                                                                                                       \
+            o.via.bin.size = size;                                                                                  \
+        }                                                                                                           \
+    };
+
+#define MSGPACK_ADAPTOR_OBJECT_WITH_ZONE_SPAN_BINARY(byte_t)                                                        \
+    template <>                                                                                                     \
+    struct object_with_zone<std::span<byte_t> > {                                                                   \
+        void operator()(msgpack::object::with_zone& o, const std::span<byte_t>& v) const {                          \
+            uint32_t size = checked_get_container_size(v.size());                                                   \
+            o.type = msgpack::type::BIN;                                                                            \
+            o.via.bin.size = size;                                                                                  \
+            if (size != 0) {                                                                                        \
+                char* ptr = static_cast<char*>(o.zone.allocate_align(size, MSGPACK_ZONE_ALIGNOF(char)));            \
+                o.via.bin.ptr = ptr;                                                                                \
+                std::memcpy(ptr, v.data(), size);                                                                   \
+            }                                                                                                       \
+        }                                                                                                           \
+    };
+
+#define MSGPACK_ADAPTOR_SPAN_BINARY(byte_t)                                                                         \
+    MSGPACK_ADAPTOR_CONVERT_SPAN_BINARY(byte_t)                                                                     \
+    MSGPACK_ADAPTOR_PACK_SPAN_BINARY(byte_t)                                                                        \
+    MSGPACK_ADAPTOR_PACK_SPAN_BINARY(const byte_t)                                                                  \
+    MSGPACK_ADAPTOR_OBJECT_SPAN_BINARY(byte_t)                                                                      \
+    MSGPACK_ADAPTOR_OBJECT_SPAN_BINARY(const byte_t)                                                                \
+    MSGPACK_ADAPTOR_OBJECT_WITH_ZONE_SPAN_BINARY(byte_t)                                                            \
+    MSGPACK_ADAPTOR_OBJECT_WITH_ZONE_SPAN_BINARY(const byte_t)
+
+MSGPACK_ADAPTOR_SPAN_BINARY(char)
+MSGPACK_ADAPTOR_SPAN_BINARY(unsigned char)
+MSGPACK_ADAPTOR_SPAN_BINARY(std::byte)
```

Comment:
I think that the macros are helping template specialization but are not used from other location.
So they can be `#undef`.

Reply:
Thank you, done.